test/e2e/suites/e2e/e2e_test.go

```diff
@@ -469,6 +471,29 @@ var _ = Describe("e2e tests [PR-Blocking]", func() {
 				return 1
 			}, e2eCtx.E2EConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Equal(1))
 			Expect(trunk.PortID).To(Equal(port.ID))
+			shared.Logf("Trunk ID: %s", trunk.ID)
+
+			// Add subports to the trunked port.
+			shared.Logf("Adding subports to the trunked port")
+			providerClient, clientOpts, _, err := shared.GetTenantProviderClient(e2eCtx)
+			Expect(err).To(BeNil(), "Cannot create providerClient")
+			networkClient, err := openstack.NewNetworkV2(providerClient, gophercloud.EndpointOpts{
+				Region: clientOpts.RegionName,
+			})
+			Expect(err).To(BeNil(), "Cannot create compute client")
+
+			addSubportsOpts := trunks.AddSubportsOpts{
+				Subports: []trunks.Subport{
+					{
+						SegmentationID:   1,
+						SegmentationType: "vlan",
+						PortID:           port.ID,
```

Comment:
It looks like you are using the `port.ID`, which is equal to `trunk.PortID` and consequently it's the Trunk parent port as the subport. So, you need to create a separate Port to be added as the subport.

Reply:
Thank you very much! That was indeed the issue!